From de8bcc303e6bcc2201c2f4f310525243118b8c18 Mon Sep 17 00:00:00 2001
From: Simon Cook <simon.cook@embecosm.com>
Date: Tue, 3 Sep 2013 09:45:04 +0100
Subject: [PATCH 12/33] * config/epiphany/epiphany.h
 (EPIPHANY_LIBRARY_EXTRA_SPEC): Don't build libraries with -fpic for now.
 (ASM_OUTPUT_SYMBOL_REF): Enable and only use @PLT for functions.

---
 gcc/ChangeLog.epiphany         | 6 ++++++
 gcc/config/epiphany/epiphany.h | 6 +++---
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/gcc/ChangeLog.epiphany b/gcc/ChangeLog.epiphany
index 8093a4519d1..85e34c7ec7d 100644
--- a/gcc/ChangeLog.epiphany
+++ b/gcc/ChangeLog.epiphany
@@ -1,3 +1,9 @@
+2013-09-03  Simon Cook  <simon.cook@embecosm.com>
+
+	* config/epiphany/epiphany.h (EPIPHANY_LIBRARY_EXTRA_SPEC): Don't
+	build libraries with -fpic for now.
+	(ASM_OUTPUT_SYMBOL_REF): Enable and only use @PLT for functions.
+
 2013-08-08  Joern Rennecke <joern.rennecke@embecosm.com>
 
 	* config/epiphany/epiphany.c (epiphany_expand_epilogue): Avoid
diff --git a/gcc/config/epiphany/epiphany.h b/gcc/config/epiphany/epiphany.h
index 42c3314f0d0..53eb1202a7e 100644
--- a/gcc/config/epiphany/epiphany.h
+++ b/gcc/config/epiphany/epiphany.h
@@ -58,7 +58,7 @@ along with GCC; see the file COPYING3.  If not see
 #define ENDFILE_SPEC "crtend.o%s crtn.o%s"
 
 #define EPIPHANY_LIBRARY_EXTRA_SPEC \
-  "-ffixed-r40 -ffixed-r41 -ffixed-r42 -ffixed-r43 -fpic"
+  "-ffixed-r40 -ffixed-r41 -ffixed-r42 -ffixed-r43"
 
 /* In the "spec:" rule,, t-epiphany changes this to epiphany_library_stub_spec
    and epiphany_library_extra_spec, respectively.  */
@@ -738,12 +738,12 @@ extern char epiphany_punct_chars[256];
 #define PRINT_OPERAND_PUNCT_VALID_P(CHAR) \
   epiphany_punct_chars[(unsigned char) (CHAR)]
 
-#if 0 /* FIXME: Assembler is not yet ready.  */
+#if 1 /* FIXME: Assembler is not yet ready.  */
 #define ASM_OUTPUT_SYMBOL_REF(FILE, SYM) \
   do \
     { \
       assemble_name ((FILE), (XSTR ((SYM), 0))); \
-      if (flag_pic && ! SYMBOL_REF_LOCAL_P (x)) \
+      if (flag_pic && ! SYMBOL_REF_LOCAL_P (x) && SYMBOL_REF_FUNCTION_P (SYM)) \
 	fputs ("@PLT", FILE); \
     } \
   while (0)
-- 
2.31.1

